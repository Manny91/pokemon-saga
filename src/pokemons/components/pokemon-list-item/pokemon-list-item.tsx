import React from "react";
import { PokemonImage } from "../pokemon-image/pokemon-image";
import styled from "../../../styled-components";

export const PokemonListItem = ({ name, id }: PokemonListItemProps) => {
  return (
    <PokemonItemWrapper>
      <PokemonItemId> {`#${id}`}</PokemonItemId>
      <PokemonItemBody>
        <PokemonImage id={id} />
        <PokemonItemTitle>{name}</PokemonItemTitle>
      </PokemonItemBody>
    </PokemonItemWrapper>
  );
};

export const PokemonItemWrapper = styled.li`
  display: flex;
  flex-direction: row;
  border: 2px solid black;
  cursor: pointer;
`;

const PokemonItemId = styled.p`
  font-size: 20px;
  padding: ${props => props.theme.spacing.sm};
`;
const PokemonItemBody = styled.div`
  display: flex;
  align-items: center;
`;
const PokemonItemTitle = styled.p`
  font-size: 16px;
  text-align: center;
  text-transform: capitalize;
  margin-left: ${props => props.theme.spacing.sm};
`;

interface PokemonListItemProps {
  name: string;
  id: string;
}
