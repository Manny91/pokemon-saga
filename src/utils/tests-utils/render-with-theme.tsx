import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
