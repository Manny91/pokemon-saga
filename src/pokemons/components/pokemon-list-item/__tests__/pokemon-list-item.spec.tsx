import React from "react";
import { PokemonListItem } from "../pokemon-list-item";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonListItem Component", () => {
  it("should render properly", () => {
    const testProps = { id: "1", name: "bulbasaur" };
    const subject = renderWithTheme(<PokemonListItem {...testProps} />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
