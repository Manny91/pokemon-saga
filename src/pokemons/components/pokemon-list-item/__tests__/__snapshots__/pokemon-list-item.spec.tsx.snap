// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonListItem Component should render properly 1`] = `
.c3 {
  image-rendering: pixelated;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border: 2px solid black;
  cursor: pointer;
}

.c1 {
  font-size: 20px;
  padding: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 16px;
  text-align: center;
  text-transform: capitalize;
  margin-left: 10px;
}

<div>
  <li
    class="c0"
  >
    <p
      class="c1"
    >
       
      #1
    </p>
    <div
      class="c2"
    >
      <img
        class="c3"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      />
      <p
        class="c4"
      >
        bulbasaur
      </p>
    </div>
  </li>
</div>
`;
